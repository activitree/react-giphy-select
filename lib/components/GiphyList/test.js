"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _index = _interopRequireDefault(require("./index"));

var _giphyData = _interopRequireDefault(require("./helpers/giphyData.json"));

describe('<GiphyList />', function () {
  test('render component', function () {
    var spy = jest.spyOn(_index.default.prototype, 'render');
    var wrapper = shallow( /*#__PURE__*/React.createElement(_index.default, {
      items: [],
      loadNextPage: function loadNextPage() {}
    }));
    expect(spy).toHaveBeenCalled();
    spy.mockReset();
    spy.mockRestore();
  });
  test('snapshot match', function () {
    var wrapper = render( /*#__PURE__*/React.createElement(_index.default, {
      items: _giphyData.default,
      loadNextPage: function loadNextPage() {}
    }));
    expect(wrapper).toMatchSnapshot();
  });
  test('set custom theme', function () {
    var customTheme = {
      list: 'customList',
      listScrollbar: 'customListScrollbar',
      listScrollbarThumb: 'customListScrollbarThumb',
      listMasonry: 'customListMasonry',
      listItem: 'customListItem',
      listEntry: 'customListEntry',
      listEntryImage: 'customListEntryImage'
    };
    var wrapper = render( /*#__PURE__*/React.createElement(_index.default, {
      items: _giphyData.default,
      loadNextPage: function loadNextPage() {},
      theme: customTheme
    }));
    expect(Object.keys(customTheme).every(function (key) {
      return wrapper.find(".".concat(customTheme[key])).length;
    })).toBe(true);
  });
  test('set custom "renderEntry" method', function () {
    var customRenderEntry = jest.fn();
    var wrapper = render( /*#__PURE__*/React.createElement(_index.default, {
      items: _giphyData.default,
      loadNextPage: function loadNextPage() {},
      renderEntry: customRenderEntry
    }));
    expect(customRenderEntry).toHaveBeenCalled();
  });
  test('set "onEntrySelect" method', function () {
    var onEntrySelect = jest.fn();
    var wrapper = mount( /*#__PURE__*/React.createElement(_index.default, {
      items: _giphyData.default,
      loadNextPage: function loadNextPage() {},
      onEntrySelect: onEntrySelect
    }));
    wrapper.find('button').simulate('click');
    expect(onEntrySelect).toHaveBeenCalled();
  });
});
//# sourceMappingURL=test.js.map