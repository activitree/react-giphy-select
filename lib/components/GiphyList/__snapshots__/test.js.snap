// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GiphyList /> snapshot match 1`] = `
<div
  class="react-giphy-select__src-components-GiphyList-styles__list___Tdg5X"
>
  <div
    style="position:relative;overflow:hidden;width:100%;height:100%;"
  >
    <div
      style="position:absolute;top:0;left:0;right:0;bottom:0;overflow:scroll;-webkit-overflow-scrolling:touch;margin-right:0;margin-bottom:0;"
    >
      <div
        class="react-giphy-select__src-components-GiphyList-styles__listMasonry___3rIoZ"
        role="listbox"
      >
        <div
          class="react-giphy-select__src-components-GiphyList-styles__listItem___2RULM"
        >
          <button
            aria-selected="true"
            class="react-giphy-select__src-components-GiphyList-styles__listEntry___D_T4j"
            role="option"
            style="width:100px;height:56px;background-image:url(http://media0.giphy.com/media/op7uqYWBm3R04/100w_s.gif);"
          >
            <img
              alt="food-krush-groove-the-fat-boys-op7uqYWBm3R04"
              class="react-giphy-select__src-components-GiphyList-styles__listEntryImage___1oyhk"
              height="56"
              src="http://media0.giphy.com/media/op7uqYWBm3R04/100w.gif"
              width="100"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      style="position:absolute;height:6px;display:none;right:2px;bottom:2px;left:2px;border-radius:3px;"
    >
      <div
        style="position:relative;display:block;height:100%;cursor:pointer;border-radius:inherit;background-color:rgba(0,0,0,.2);"
      />
    </div>
    <div
      class="react-giphy-select__src-components-GiphyList-styles__listScrollbar___1dS3n"
    >
      <div
        class="react-giphy-select__src-components-GiphyList-styles__listScrollbarThumb___1L0ZG"
        style="position:relative;display:block;width:100%;"
      />
    </div>
  </div>
</div>
`;
